tools/build/build.js

```diff
@@ -21,39 +21,84 @@ if (NODE_VERSION < NODE_VERSION_TARGET) {
   process.exit(1);
 }
 
+const STANDARD_BUILD = "Standard Build"
+const TGS_BUILD = "TGS Build"
+const ALL_MAPS_BUILD = "CI All Maps Build"
+const TEST_RUN_BUILD = "CI Integration Tests Build"
+
+let BUILD_MODE = STANDARD_BUILD;
+if(process.env.CBT_BUILD_MODE != undefined){
+  switch (process.env.CBT_BUILD_MODE) {
+    case "ALL_MAPS":
+      BUILD_MODE = ALL_MAPS_BUILD
+      break;
+    case "TEST_RUN":
+      BUILD_MODE = TEST_RUN_BUILD
+      break;
+    case "TGS":
+      BUILD_MODE = TGS_BUILD
+      break;
+    default:
+      BUILD_MODE = process.env.CBT_BUILD_MODE
+      break;
+  }
+}
+console.log(`Starting CBT in ${BUILD_MODE} mode.`)
+
+const DME_NAME = 'tgstation'
+
 // Main
 // --------------------------------------------------------
 
 const { resolveGlob, stat } = require('./cbt/fs');
 const { exec } = require('./cbt/process');
 const { Task, runTasks } = require('./cbt/task');
 const { regQuery } = require('./cbt/winreg');
+const fs = require('fs');
+
+const yarn = args => {
+  const yarnPath = resolveGlob('./tgui/.yarn/releases/yarn-*.cjs')[0]
+    .replace('/tgui/', '/');
+  return exec('node', [yarnPath, ...args], {
+    cwd: './tgui',
+  });
+};
+
+/// Installs all tgui dependencies
+const taskYarn = new Task('yarn')
+  .build(async () => {
+    await yarn(['install']);
+  });
+
+/// Builds svg fonts
+const taskTgfont = new Task('tgfont')
+  .depends('tgui/.yarn/install-state.gz')
+  .depends('tgui/packages/tgfont/**/*.+(js|cjs|svg)')
+  .depends('tgui/packages/tgfont/package.json')
+  .provides('tgui/packages/tgfont/dist/tgfont.css')
+  .provides('tgui/packages/tgfont/dist/tgfont.eot')
+  .provides('tgui/packages/tgfont/dist/tgfont.woff2')
+  .build(async () => {
+    await yarn(['workspace', 'tgfont', 'build']);
+  });
 
+/// Builds tgui
 const taskTgui = new Task('tgui')
-  .depends('tgui/.yarn/releases/*')
   .depends('tgui/.yarn/install-state.gz')
-  .depends('tgui/yarn.lock')
   .depends('tgui/webpack.config.js')
   .depends('tgui/**/package.json')
-  .depends('tgui/packages/**/*.+(js|jsx|ts|tsx|cjs|mjs|scss)')
+  .depends('tgui/packages/**/*.+(js|jsx|ts|tsx|cjs|scss)')
   .provides('tgui/public/tgui.bundle.css')
   .provides('tgui/public/tgui.bundle.js')
   .provides('tgui/public/tgui-common.bundle.js')
   .provides('tgui/public/tgui-panel.bundle.css')
   .provides('tgui/public/tgui-panel.bundle.js')
   .build(async () => {
-    // Instead of calling `tgui/bin/tgui`, we reproduce the whole pipeline
-    // here for maximum compilation speed.
-    const yarnPath = resolveGlob('./tgui/.yarn/releases/yarn-*.cjs')[0]
-      .replace('/tgui/', '/');
-    const yarn = args => exec('node', [yarnPath, ...args], {
-      cwd: './tgui',
-    });
-    await yarn(['install']);
     await yarn(['run', 'webpack-cli', '--mode=production']);
   });
 
-const taskDm = new Task('dm')
+const taskDm = (...injectedDefines) => {
+   return new Task('dm')
```

Comment:
Can you raise the indentation levels on the `.depends` stuff, since we're going up a layer?